lib/utils/data_service.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:convert';
+import 'package:http/http.dart';
+
+class DataService {
```

Comment:
Error handling would probably be a good idea here. A common pattern when using `get` is to wrap it in a try/catch block. That way if there is not internet connection, bad signal, or an error is returned from the service, you can catch it and handle it.

Reply:
I actually had a try/catch block but I wanted to check how `snapshot.hasError` and `snapshot.hasData` worked and forgot to add it back 😅